src/core/simulator/simulator.cpp

```diff
@@ -379,7 +379,7 @@ namespace argos {
                                          "length",
                                          fExpLength,
                                          0.0f);
-         m_unMaxSimulationClock = fExpLength * unTicksPerSec;
+         m_unMaxSimulationClock = static_cast<UInt32>(fExpLength * unTicksPerSec);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a implicit float -> int conversion warning without the cast